crates/shared/src/request_sharing.rs

```diff
@@ -86,9 +96,18 @@ where
         });
 
         if let Some(existing) = existing {
+            Metrics::get()
+                .request_sharing_access
+                .with_label_values(&[&self.request_label, "hits"])
+                .inc();
             return existing;
         }
 
+        Metrics::get()
+            .request_sharing_access
+            .with_label_values(&[&self.request_label, "misses"])
+            .inc();
```

Comment:
I think these metrics on their own might not tell you what you want to know.
For example if some bot always fetches `optimal` quotes no request sharing will ever happen. It could look like this caching mechanism is insufficient. I don't have a good suggestion to remove that bias but thought I should point it out to aid the evaluation of the results.